modules/xdcr-reference/pages/xdcr-filtering-data-type-conversion.adoc

```diff
@@ -0,0 +1,479 @@
+
+| 14
+| `<Any-Except-Regex>`
+| `Regex`
+| image:xdcr-reference:cross-mark-red-sm-wide.png[] Invalid.
+| Collation comparison is performed for all except `Regex`.
+
+| 15
+| `<Any-Except-Pcre>`
+| `Pcre`
+| image:xdcr-reference:cross-mark-red-sm-wide.png[] Invalid.
+| Collation comparison is performed for all except `Pcre`.
+
+| 16
+| `<Any-Except-Null>`
+| `Null`
+| image:xdcr-reference:checkmark-green-sm.png[] Valid.
+`<Any>` is converted a `non-Null` value, for comparison with `Null`.
+| The comparison `non-Null > Null` returns `true`.
+
+| 17
+| `Int`
+| `Boolean`
+| image:xdcr-reference:checkmark-green-sm.png[] Valid for all `Int` values.
+The `Int` value `0` is converted to the `Boolean` value `false`; all other `Int` values are converted to the `Boolean` value `true`.
+| The boolean-comparison `true > false` returns `true`.
+
+| 18
+| `Uint`
+| `Boolean`
+| image:xdcr-reference:checkmark-green-sm.png[] Valid for all `Uint` values.
+The `Uint` value `0` is converted to the `Boolean` value `false`; all other `Uint` values are converted to the `Boolean` value `true`.
+| The boolean-comparison `true > false` returns `true`.
+
+| 19
+| `Float`
+| `Boolean`
+| image:xdcr-reference:checkmark-green-sm.png[] Valid for all `Float` values.
+The `Float` value `0.0` is converted to the `Boolean` value `false`; all other `Float` values are converted to the `Boolean` value `true`.
+| The boolean-comparison `true > false` returns `true`.
+
+| 20
+| `String`
+| `Boolean`
+| image:xdcr-reference:checkmark-orange-sm.png[] Valid if `String` is case-insensitive `"true"` or `"false"`; in which case `String` is converted to its `Boolean` equivalent.
+| The boolean-comparison `true > false` returns `true`.
+
+| 21
+| `<Any-Except-Array>`
+| `Array`
+| image:xdcr-reference:cross-mark-red-sm-wide.png[] Invalid.
+| Collation comparison is performed.
+
+| 22
+| `<Any-Except-Object>`
+| `Object`
+| image:xdcr-reference:cross-mark-red-sm-wide.png[] Invalid.
+| Collation comparison is performed. is used.
+
+| 23
+| `<Any-Except-Time-and-String`
+| `Time`
+| image:xdcr-reference:cross-mark-red-sm-wide.png[] Invalid.
+| Collation comparison is performed.
+
+| 24
+| `String`
+| `Time`
+| image:xdcr-reference:checkmark-orange-sm.png[] Valid if `String` can be parsed as a parameter to the `DATE` function.
+| Standard comparision is performed if valid; otherwise, collation comparison is performed.
+
+| 25
+| `Array`
+| `Array`
+| image:xdcr-reference:dash-black-inline.png[] No conversion need be performed.
+| Comparison is performed along the following lines: if the length of `Array_1` is greater than the length of `Array_2`, then `Array_1 > Array_2` returns `true`.
+If the arrays are of equal length, corresponding objects are compared in sequence, until one is found to be the greater, and `true` is returned &#8212; _i.e._ `Array_1[object_1] > Array_2[object_1]`, `Array_1[object_2] > Array_2[object_2]`, and so forth.
```

Comment:
> until one is found to be the greater, and `true` is returned

The above reads as if "greater" and "true" are the operations being used internally, which may not be generic enough.

The idea to convey here should be objects within the arrays are compared in sequence against the expression set forth by the user if they are of the same length. What if the user had the expression `arr1 != arr2 == true` and the arrays are of the same length and only the last object of both arrays are different? The algorithm here is to compare objects by object until it figures out arr1 is > or < than arr2 due to the last element within the array, and then apply the operation "!=" to the result, which is true.

Reply:
Would the following be better? 

Comparison is performed along the following lines: the specified expression is first applied to the arrays themselves, based on the respective array-lengths.
For example, if the expression is `arg_1 > arg_2`, this gets applied as `Array_1 > Array_2`; and if the length of `Array_1` is indeed greater than the length of `Array_2`, the requirement is considered to have been met; and the comparison-process ends.

In cases where the requirement is _not_ met based on array-lengths, due to the array-lengths being _equal_, the comparison-process continues as follows: in sequence, pairs of correspondingly positioned objects from the arrays are compared, until the specified requirement is met, or is finally determined _not_ to have been met.
For example, if the expression is `arg_1 > arg_2`, this gets applied as `Array_1[object_1] > Array_2[object_1]`, `Array_1[object_2] > Array_2[object_2]`, and so forth.

Question: So, if the condition not met because Array_1 < Array_2, does that mean the result is false, and comparison ends at that point?

Reply:
> For example, if the expression is arg_1 > arg_2, this gets applied as Array_1[object_1] > Array_2[object_1], Array_1[object_2] > Array_2[object_2], and so forth.

Arrays are indexed by positive integers and not a specific object so it may be more technically correct to say "Array_1[0] > Array_2[0]". 
I understand the concept being conveyed but it looked kinda odd from a coding standpoint.

> Question: So, if the condition not met because Array_1 < Array_2, does that mean the result is false, and comparison ends at that point?

I'm assuming you're asking 
> if the condition is not met because the *length* of Array_1 is smaller than the length of Array_2 when the expression given is 'Array_1 > Array_2'
Then yes, the result returned is "false" and the object-by-object comparison is not executed as it isn't needed.

Reply:
Ok. Thanks. Will change to [0] etc.